sympy/physics/control/lti.py

```diff
@@ -1422,3 +1445,563 @@ def _eval_rewrite_as_TransferFunction(self, num, den, **kwargs):
+        >>> tf1 = TransferFunction(p**2 - 1, s**4 + s**3 - p, p)
+        >>> tf2 = TransferFunction(1 - p, p**2 - 3*p + 7, p)
+        >>> tf3 = TransferFunction(3, 4, p)
+        >>> tfm1 = TransferFunctionMatrix([[tf1, -tf2]])
+        >>> tfm1.shape
+        (1, 2)
+        >>> tfm2 = TransferFunctionMatrix([[-tf2, tf3], [tf1, -tf1]])
+        >>> tfm2.shape
+        (2, 2)
+
+        """
+        return self._expr_mat.shape
+
+    def __neg__(self):
+        neg = -self._expr_mat
+        return _to_TFM(neg, self.var)
+
+    def __getitem__(self, key):
+        trunc = self._expr_mat.__getitem__(key)
+        if isinstance(trunc, ImmutableMatrix):
+            return _to_TFM(trunc, self.var)
+        return TransferFunction.from_rational_expression(trunc, self.var)
+
+    def transpose(self):
+        """Returns the transpose of the ``TransferFunctionMatrix`` (switched input and output layers)."""
+        transposed_mat = self._expr_mat.transpose()
+        return _to_TFM(transposed_mat, self.var)
+
+    def doit(self):
+        """
+        Returns the equivalent ``TransferFunctionMatrix`` with each ``Series`` and ``Parallel``
+        element rewritten as ``TransferFunction``.
+
+        Examples
+        ========
+
+        >>> from sympy.abc import s
+        >>> from sympy.physics.control.lti import TransferFunction, TransferFunctionMatrix, Series, Parallel
+        >>> from sympy.printing import pprint
+        >>> tf_1 = TransferFunction(1, s, s)
+        >>> tf_2 = TransferFunction(1, s**2, s)
+        >>> tfm_1 = TransferFunctionMatrix([[Series(tf_1, tf_2), tf_1], [tf_2, Parallel(tf_1, tf_2)]])
+        >>> tfm_1
+        TransferFunctionMatrix(((Series(TransferFunction(1, s, s), TransferFunction(1, s**2, s)), TransferFunction(1, s, s)), (TransferFunction(1, s**2, s), Parallel(TransferFunction(1, s, s), TransferFunction(1, s**2, s)))))
+        >>> pprint(tfm_1, use_unicode=False)
+        [ 1      1   ]
+        [----    -   ]
+        [   2    s   ]
+        [s*s         ]
+        [            ]
+        [ 1    1    1]
+        [ --   -- + -]
+        [  2    2   s]
+        [ s    s     ]
+        >>> tfm_1.doit()
+        TransferFunctionMatrix(((TransferFunction(1, s**3, s), TransferFunction(1, s, s)), (TransferFunction(1, s**2, s), TransferFunction(s**2 + s, s**3, s))))
+        >>> pprint(_, use_unicode=False)
+        [1     1   ]
+        [--    -   ]
+        [ 3    s   ]
+        [s         ]
+        [          ]
+        [     2    ]
+        [1   s  + s]
+        [--  ------]
+        [ 2     3  ]
+        [s     s   ]
+
+        """
+        doit_arg = []
+        for i in self.args[0]:
+            temp = []
+            for j in i:
+                if not isinstance(j, TransferFunction):
+                    j = j.doit()
+                temp.append(j)
+            doit_arg.append(temp)
+        return TransferFunctionMatrix(doit_arg)
```

Comment:
This code is somewhat convoluted:
```
arg = [[element.doit() for element in row] for row in self.args[0]]
return TransferFunctionMatrix(arg)
```
Then again is that even needed?

Surely the `Basic.doit` method already handles this.